### PR TITLE
Awesome Documentation & Automation

### PR DESCRIPTION
**Pull request by jGleitz**
Over the christmas holidays, I took my time to build us ourselves a web presence and heavily improve our build logic and documentation.

### Gradle
 * Build logic is split and lives in the `buildSrc` subproject
 * The new system is build to be scalable and support good modularisation for an ever growing build logic code base

### Web Page
 * I build [a small web page](http://jgleitz.github.io/Beagle/branches/awesome-automation/) that represents our project. 
 * Apart from being informative, it serves as an overview to all build artefacts. As the web page is build for all branches, we can link to it in Pull Requests to give reviewers quick access to the javadoc, checkstyle report, requirements specification, etc.
 * I used [Handlebars](http://handlebarsjs.com/) as an template engine. Pages can be written in Markdown. Thus everybody can easily edit pages and add new ones.

### Documentation
 * I added and edited `README.md` files throughout our project to better document our project structure.
 * I fixed some javadoc issues in the core stubs.

##### Why did you spend so much time on this? There are surely more important tasks at the moment!
First of all, if I worked over the christmas holidays and between Christmas and New Year, I wanted to do something fun. And building a website and discovering gradle was a lot of fun. Second, I truly think great documentation is crucial if we wand developers to continue our project. And I learned a lot along the way. Gradle is an *awesome* tool. But you need some time to get to know all of its features.

##### Why the `buildSrc`? Can’t we use `src/build` or something like that?
It’s a gradle convention and makes things a lot easier.

##### Why are you taking extra effort to pretty print the html output?
handlebars.java creates html that looks ugly (wrong indentation and newlines all around the place). Pretty printing the output makes debugging a lot easier and you can actually start reading the HTML

##### Couldn’t you optimize the web page?
I could. As a matter of fact, I wrote a web page that is optimised using all contemporary techniques available. The results are *stunning*.
But optimisation is a huge task that makes the build logic very complex. I decided for a structure that is easy to maintain and easy to understand. This includes a lot of decisions that are horrible for performance.

Unfortunately, the web fonts don’t look as great because they lack metaness (which is only supported by metafont). But I still think it looks good!

[Review on Reviewable](https://reviewable.io/reviews/beagle-pse/beagle/353)

### CODE REVIEW

**Comment by acBerger** [comment]
Reviewed 51 of 51 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[buildSrc/src/main/groovy/web/BeagleHelpers.groovy, line 2 \[r1\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6soA2pspw-cRLYNc4Z:-K6soA2q2mBDSd_7xDjC:485480827) ([raw file](https://github.com/beagle-pse/beagle/blob/e6ba0fdfa0798ce088e5f89a6fd9d18b8bc8dc17/buildSrc/src/main/groovy/web/BeagleHelpers.groovy#L2)):*
> Helpers

should be 'helpers'

---

*[buildSrc/src/main/groovy/web/BeagleHelpers.groovy, line 10 \[r1\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6soHDOSloDMyZS06U_:-K6soHDOSloDMyZS06Ua:857248772) ([raw file](https://github.com/beagle-pse/beagle/blob/e6ba0fdfa0798ce088e5f89a6fd9d18b8bc8dc17/buildSrc/src/main/groovy/web/BeagleHelpers.groovy#L10)):*
'helpers'

---

*[buildSrc/src/main/groovy/web/HTMLFormatFilter.groovy, line 6 \[r1\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6soqNC3DQM2F6zLTIn:-K6soqNC3DQM2F6zLTIo:-1775970878) ([raw file](https://github.com/beagle-pse/beagle/blob/e6ba0fdfa0798ce088e5f89a6fd9d18b8bc8dc17/buildSrc/src/main/groovy/web/HTMLFormatFilter.groovy#L6)):*
'reader'

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353)*

**Comment by m1cm1c** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[buildSrc/src/main/groovy/web/BeagleHelpers.groovy, line 11 \[r1\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6sqI99gMf2WkmeUIws:-K6sqI99gMf2WkmeUIwt:-468874064) ([raw file](https://github.com/beagle-pse/beagle/blob/e6ba0fdfa0798ce088e5f89a6fd9d18b8bc8dc17/buildSrc/src/main/groovy/web/BeagleHelpers.groovy#L11)):*
'instance'

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353)*

**Comment by jGleitz** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[buildSrc/src/main/groovy/web/HTMLFormatFilter.groovy, line 6 \[r1\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6soqNC3DQM2F6zLTIn:-K6sqL_ufasffejo0vu3:-1610166478) ([raw file](https://github.com/beagle-pse/beagle/blob/e6ba0fdfa0798ce088e5f89a6fd9d18b8bc8dc17/buildSrc/src/main/groovy/web/HTMLFormatFilter.groovy#L6)):*
It’s a name here (because “FilterReader” is an API class).

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353)*

**Comment by m1cm1c** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[buildSrc/src/main/groovy/web/BeagleHelpers.groovy, line 31 \[r1\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6sqSGyFymm9Hj9LHtP:-K6sqSGyFymm9Hj9LHtQ:-774950089) ([raw file](https://github.com/beagle-pse/beagle/blob/e6ba0fdfa0798ce088e5f89a6fd9d18b8bc8dc17/buildSrc/src/main/groovy/web/BeagleHelpers.groovy#L31)):*
'an' → 'a'

---

*[buildSrc/src/main/groovy/web/BeagleHelpers.groovy, line 34 \[r1\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6sqWj-em6fpWwfkw-J:-K6sqWj-em6fpWwfkw-K:-774950089) ([raw file](https://github.com/beagle-pse/beagle/blob/e6ba0fdfa0798ce088e5f89a6fd9d18b8bc8dc17/buildSrc/src/main/groovy/web/BeagleHelpers.groovy#L34)):*
'an' → 'a'

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353)*

**Comment by acBerger** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[buildSrc/src/main/groovy/web/HTMLFormatFilter.groovy, line 6 \[r1\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6soqNC3DQM2F6zLTIn:-K6sqf9BmepN09fuwEOb:-1698239107) ([raw file](https://github.com/beagle-pse/beagle/blob/e6ba0fdfa0798ce088e5f89a6fd9d18b8bc8dc17/buildSrc/src/main/groovy/web/HTMLFormatFilter.groovy#L6)):*
okay, I didn't get that before. My bad.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353)*

**Comment by m1cm1c** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[buildSrc/src/main/groovy/web/HTMLFormatFilter.groovy, line 6 \[r1\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6soqNC3DQM2F6zLTIn:-K6sqx7F02m2JCUJDT_Y:-832479700) ([raw file](https://github.com/beagle-pse/beagle/blob/e6ba0fdfa0798ce088e5f89a6fd9d18b8bc8dc17/buildSrc/src/main/groovy/web/HTMLFormatFilter.groovy#L6)):*
'easy' → 'easily'

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353)*

**Comment by jGleitz** [comment]
Review status: 47 of 49 files reviewed at latest revision, 6 unresolved discussions.

---

*[buildSrc/src/main/groovy/web/BeagleHelpers.groovy, line 2 \[r1\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6soA2pspw-cRLYNc4Z:-K6t-xZP3qChJA-HPQ6w:676635175) ([raw file](https://github.com/beagle-pse/beagle/blob/e6ba0fdfa0798ce088e5f89a6fd9d18b8bc8dc17/buildSrc/src/main/groovy/web/BeagleHelpers.groovy#L2)):*
Done

---

*[buildSrc/src/main/groovy/web/BeagleHelpers.groovy, line 10 \[r1\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6soHDOSloDMyZS06U_:-K6t0-Zhy0hhghv1f18U:676635175) ([raw file](https://github.com/beagle-pse/beagle/blob/e6ba0fdfa0798ce088e5f89a6fd9d18b8bc8dc17/buildSrc/src/main/groovy/web/BeagleHelpers.groovy#L10)):*
Done

---

*[buildSrc/src/main/groovy/web/BeagleHelpers.groovy, line 11 \[r1\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6sqI99gMf2WkmeUIws:-K6t0APfBYPXtkbLI5XA:676635175) ([raw file](https://github.com/beagle-pse/beagle/blob/e6ba0fdfa0798ce088e5f89a6fd9d18b8bc8dc17/buildSrc/src/main/groovy/web/BeagleHelpers.groovy#L11)):*
Done

---

*[buildSrc/src/main/groovy/web/BeagleHelpers.groovy, line 31 \[r1\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6sqSGyFymm9Hj9LHtP:-K6t03TeIrHWZvWxsoOu:676635175) ([raw file](https://github.com/beagle-pse/beagle/blob/e6ba0fdfa0798ce088e5f89a6fd9d18b8bc8dc17/buildSrc/src/main/groovy/web/BeagleHelpers.groovy#L31)):*
Done

---

*[buildSrc/src/main/groovy/web/BeagleHelpers.groovy, line 34 \[r1\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6sqWj-em6fpWwfkw-J:-K6t05KuDe1UN0e34vnV:676635175) ([raw file](https://github.com/beagle-pse/beagle/blob/e6ba0fdfa0798ce088e5f89a6fd9d18b8bc8dc17/buildSrc/src/main/groovy/web/BeagleHelpers.groovy#L34)):*
Done

---

*[buildSrc/src/main/groovy/web/HTMLFormatFilter.groovy, line 6 \[r1\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6soqNC3DQM2F6zLTIn:-K6t0VAQ4Ku1gVl-uewC:676635175) ([raw file](https://github.com/beagle-pse/beagle/blob/e6ba0fdfa0798ce088e5f89a6fd9d18b8bc8dc17/buildSrc/src/main/groovy/web/HTMLFormatFilter.groovy#L6)):*
Done

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353)*

**Comment by acBerger** [comment]
:lgtm: 

---

Reviewed 1 of 1 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-:-K6t0tdf0STccxkAy2a5:980077423)*

**Comment by m1cm1c** [comment]
Reviewed 49 of 51 files at r1, 1 of 1 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/web/handlebars/pages/README.md, line 3 \[r3\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6tFrCxT5_fuotXT7db:-K6tFrCxT5_fuotXT7dc:1119035337) ([raw file](https://github.com/beagle-pse/beagle/blob/355d4d36d4da0eba7b94c833e772ac85af327bf5/src/web/handlebars/pages/README.md#L3)):*
This needs to be filled in. I don't see anything filling `publishUrl` in automatically, so I assume it has to be done manually.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353)*

**Comment by jGleitz** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/web/handlebars/pages/README.md, line 3 \[r3\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6tFrCxT5_fuotXT7db:-K6xP_tMuw5368pYY4FN:-1530782077) ([raw file](https://github.com/beagle-pse/beagle/blob/355d4d36d4da0eba7b94c833e772ac85af327bf5/src/web/handlebars/pages/README.md#L3)):*
As detailed in [the README](https://github.com/jGleitz/Beagle/tree/awesome-automation/src/web), all files in the `pages` folder will be copied into the web presence (e.g. the GitHub pages branch) and handlebars templates will be rendered.  `{{publishUrl}}` is [a helper defined in `BeagleHelpers.groovy`](https://github.com/jGleitz/Beagle/blob/awesome-automation/buildSrc/src/main/groovy/web/BeagleHelpers.groovy#L63). There is no syntactic difference between a parameterless helper and a data field in handlebars.

[In my repository](https://github.com/jGleitz/Beagle/tree/gh-pages/branches/awesome-automation), you can see that on Travis, `{{publishUrl}}` will be correctly resolved to the URL where the current branche’s web presence will be published.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353)*

**Comment by m1cm1c** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/web/handlebars/pages/README.md, line 3 \[r3\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6tFrCxT5_fuotXT7db:-K6xW0fmK7A9Sb8s2x6H:-1191610227) ([raw file](https://github.com/beagle-pse/beagle/blob/355d4d36d4da0eba7b94c833e772ac85af327bf5/src/web/handlebars/pages/README.md#L3)):*
So it works correctly when it's replaced by an empty string on my machine?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353)*

**Comment by jGleitz** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/web/handlebars/pages/README.md, line 3 \[r3\]](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-K6tFrCxT5_fuotXT7db:-K6xX70PsB12IAN_PSCh:-2058402932) ([raw file](https://github.com/beagle-pse/beagle/blob/355d4d36d4da0eba7b94c833e772ac85af327bf5/src/web/handlebars/pages/README.md#L3)):*
Yes. The helper expects certain Environments Variables containing information about the current repository normally set by Travis. If it doesn’t find them, it’ll return an empty string.

If you want to try it out on your machine, just set the ENVs `TRAVIS_BRANCH` and `TRAVIS_REPO_SLUG`. Note that the code expects the latter to contain a slash (`/`). 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353)*

**Comment by m1cm1c** [comment]
Why is this PR assigned to me?

**Comment by jGleitz** [comment]
> Why is this PR assigned to me?

You reviewed it, so I thought it’s awaiting your response.

---

Review status: 48 of 49 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-:-K6xhXVvsF0Mz7w3RhrE:-1930637227)*

**Comment by m1cm1c** [comment]
okay

---

Review status: 48 of 49 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-:-K6xhyXMu7n3R69BvuqR:677911961)*

**Comment by m1cm1c** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353)*

**Comment by jGleitz** [comment]
@acBerger lgtm went stale because I merged.

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-:-K6xiCn88cWIB0E7LXvW:2120694540)*

**Comment by m1cm1c** [comment]
:lgtm: 

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-:-K6xidIpRXaZU5x8NCC6:980077423)*

File: buildSrc/src/tasks/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# build configuration
+All gradle files in this folder (but not in subdirectories) will be added to the gradle build configuration. They will be added in the order indicated by their file name, that’s why numbers are used to order them. The files should be kept at a reasonable small size. If more complicated logic is required, it should be put in groovy classes.
+
+## task dependencies
```
Comment (RomanLangrehr) [inline buildSrc/src/tasks/README.md]: Can you use capital letters for the titles, like in all other docs, too, please?

**Comment by acBerger** [comment]
:lgtm: 

---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-:-K6xjZjyRI_2oEFSpevt:980077423)*

**Comment by m1cm1c** [comment]
:lgtm_cancel:

**Comment by m1cm1c** [comment]
:lgtm_cancel: 

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-:-K6xjj0CRhN4TbzZtkAY:502544406)*

File: buildSrc/src/main/groovy/web/BeagleWebContext.groovy [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	 */
+	private static Task task(final String name) {
+		Set<Task> allTasks = project.getTasksByName name, true
+		return allTasks.iterator()[0]
+	}
+	
+	/** 
+	 * Makes a given path relative to the web output folder.
+	 *
+	 * @param path	A path (anything accepte by Gradles Project#files).
```
Comment (m1cm1c) [inline buildSrc/src/main/groovy/web/BeagleWebContext.groovy]: 'accepte' → 'accepted'

**Comment by RomanLangrehr** [comment]
I'd like to review this, too, because I think nobody with an windows-PC reviewed it right now.

File: buildSrc/src/main/groovy/web/BeagleWebContext.groovy [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+			/**
+			 * Computer Modern Sans, as downloaded from http://checkmyworking.com/cm-web-fonts/
+			 */
+			String cmSans = relative task('copyWebFonts').property('cmsansdest')
+		}
+	}
+	
+	/**
+	 * Returns one task on {@link #project} named {@code name}. If there are multiple such tasks,
+	 * one is picked. It is neithor defined nor reliable which one will be picked.
```
Comment (m1cm1c) [inline buildSrc/src/main/groovy/web/BeagleWebContext.groovy]: 'neithor' → 'neither'

File: buildSrc/src/main/groovy/web/BeagleWebFilter.groovy [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import org.apache.tools.ant.filters.BaseFilterReader
+import org.apache.commons.io.IOUtils 
+
+/**
+ * Abstract class to quickly create FilterReaders that operate on a whole file content string.
```
Comment (m1cm1c) [inline buildSrc/src/main/groovy/web/BeagleWebFilter.groovy]: 'whole' → 'entire'

I've seen this a lot from you and @RomanLangrehr . http://english.stackexchange.com/questions/9615/whole-vs-entire

File: buildSrc/src/main/groovy/web/BeagleWebFilter.groovy [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+ * Implementing classes should add their logic only to {@link #getResultString()}.
+ *
+ * @author Joshua Gleitze
+ */
+abstract class BeagleWebFilter extends BaseFilterReader {
+	/**
+	 * The whole input file’s content as a string.
```
Comment (m1cm1c) [inline buildSrc/src/main/groovy/web/BeagleWebFilter.groovy]: 'whole' → 'entire'

File: buildSrc/src/main/groovy/web/BeagleWebFilter.groovy [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+	 */
+	protected final String inputString
+	/**
+	 * The whole input file’s conten as a reader.
```
Comment (m1cm1c) [inline buildSrc/src/main/groovy/web/BeagleWebFilter.groovy]: 'whole' → 'entire'

File: build.gradle [hunk 7]
```diff
@@ -1,122 +1,32 @@
...
+	fileTree("$projectDir/buildSrc/src/tasks/gradle/buildscript-dependencies").include('*.gradle').each { file ->
+	     apply from: file, to: buildscript
+	} 
 }
 
-dependencies {
-	testCompile group: 'junit', name: 'junit', version: '4.+'
+/**
+ * Allows scripts to run their tasks on the main (build) task.
+ * Scripts should no add their tasks to the build tasks themselves but use this method
```
Comment (m1cm1c) [inline build.gradle]: 'no' → 'not'

File: buildSrc/src/main/groovy/web/BeagleWebFilter.groovy (hunk 4 above)
Comment (jGleitz) [inline buildSrc/src/main/groovy/web/BeagleWebFilter.groovy]: Thanks for the link!

**Comment by jGleitz** [comment]
I adopted all your comments from GitHub.

---

Review status: 40 of 49 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-:-K6xpH9j7CsdkCWyq0dL:-1108928448)*

**Comment by RomanLangrehr** [comment]
:lgtm:

**Comment by RomanLangrehr** [comment]
Reviewed 39 of 51 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, 1 of 1 files at r4, 6 of 6 files at r5, 3 of 3 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353)*

**Comment by acBerger** [comment]
:lgtm: 

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-:-K6y-sOB_GHGIYAiUrpk:980077423)*

**Comment by acBerger** [comment]
Reviewed 6 of 6 files at r5, 3 of 3 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353)*

**Comment by jGleitz** [comment]
Reviewed 1 of 51 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353)*

**Comment by jGleitz** [comment]
https://github.com/Reviewable/Reviewable/issues/224

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-:-K6y6ClxQ9Qrxmv3M8Uk:442519460)*

**Comment by acBerger** [comment]
:lgtm: 

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/353#-:-K6yVs7LaYzCSiECWkhh:980077423)*

**Pull request merged**